cmd/ferretdb/main.go

```diff
@@ -220,7 +220,18 @@ func main() {
 	case "run":
 		run()
 	case "ping":
-		ping()
+		logger := setupLogger(cli.Log.Format, "")
+		checkFlags(logger)
+
+		ready := ReadyZ{
+			l: logger,
+		}
+
+		ctx, _ := ctxutil.SigTerm(context.Background())
+		if !ready.Probe(ctx) {
+			os.Exit(1)
```

Comment:
Elsewhere, we exit with non zero error code using `logger.Fatal`, why is this special?

Reply:
In which case we use `logger.Fatal` instead of `os.Exit` when we don't return the error?

The error is already logged inside of the `Probe` function, hence nothing can be returned here. If `Probe` returns false, it means that the readiness check didn't perform correctly, so the caller can call os.Exit(1).

Reply:
Let me see if I understand your question :thinking:. We don't call os.Exit in our code base directly. Something such as `Fatal("Ping failed")` gives user friendly message on what has caused exit 1.